### PR TITLE
boards: add support for im880b lora board

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR adds [im880b](https://www.wireless-solutions.de/products/radiomodules/im880b-l.html) board, this board has a stm32l152cb microcontroler and a sx1272 radio. 

### Testing procedure

Run all tests:
`python dist/tools/compile_and_test_for_board/compile_and_test_for_board.py --jobs 0 ..RIOT im880b`

Manually test:
Compile and test the loramac test application.
`make BOARD=im880b LORA_DRIVER=sx1272-C tests/pkg_semtech-loramac clean all -j flash`

### Issues/PRs references

Depends on: ~~#11314 & #11316~~

### CODE REVIEW

**Comment by kYc0o** [comment]
Can you check if this board applies for code reusage based on the lobaro-lorabox board?

**Comment by kYc0o** [comment]
Well actually I got one of these and I'm able to test. Will check the code soon.

**Comment by kYc0o** [comment]
Please rebase if possible.

**Comment by kYc0o** [comment]
Just tested it. Works as expected.

I just didn't test the flashing since it seems to flash the same way as the Lobaro board.

File: boards/im880b/Makefile.features [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Put defined MCU peripherals here (in alphabetical order)
+FEATURES_PROVIDED += periph_adc
+FEATURES_PROVIDED += periph_i2c
+FEATURES_PROVIDED += periph_rtc
+FEATURES_PROVIDED += periph_spi
+FEATURES_PROVIDED += periph_timer
+FEATURES_PROVIDED += periph_uart
+
+# The board MPU family (used for grouping by the CI system)
```
Comment (aabadie) [inline boards/im880b/Makefile.features]: You don't need these 2 lines anymore

File: boards/im880b/Makefile.features [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Put defined MCU peripherals here (in alphabetical order)
```
Comment (aabadie) [inline boards/im880b/Makefile.features]: You should define CPU and CPU_MODEL here.

File: boards/im880b/Makefile.include [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+export CPU_MODEL = stm32l151cb
+
+# define the default port depending on the host OS
+PORT_LINUX ?= /dev/ttyUSB0
+PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.SLAB_USBtoUART*)))
+
+# setup serial terminal
+include $(RIOTMAKE)/tools/serial.inc.mk
+
+export DEBUG_ADAPTER ?= stlink
```
Comment (aabadie) [inline boards/im880b/Makefile.include]: export not needed

File: boards/im880b/Makefile.include [hunk 4]
```diff
@@ -0,0 +1,18 @@
+## the cpu to build for
+export CPU = stm32l1
```
Comment (aabadie) [inline boards/im880b/Makefile.include]: Should be moved to Makefile.features (without the export ;) )

File: boards/im880b/doc.txt [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+| Datasheet  | [Datasheet](https://www.st.com/resource/en/datasheet/stm32l151cb-a.pdf) |
+| Reference Manual | [Reference Manual](https://www.st.com/content/ccc/resource/technical/document/reference_manual/cc/f9/93/b2/f0/82/42/57/CD00240193.pdf/files/CD00240193.pdf/jcr:content/translations/en.CD00240193.pdf) |
+| Programming Manual | [Programming Manual](https://www.st.com/content/ccc/resource/technical/document/programming_manual/5b/ca/8d/83/56/7f/40/08/CD00228163.pdf/files/CD00228163.pdf/jcr:content/translations/en.CD00228163.pdf) |
+| Board Manual   | [Board Manual](https://cdn.sos.sk/productdata/29/eb/a68245ed/im880b-l-lorawan.pdf)|
+
+### User Interface
+
+## Flashing
+As no usb connector is present on the device an external programmer must
+be connected to the board for flasing. You can use JTAG or STLINK programmer.
```
Comment (aabadie) [inline boards/im880b/doc.txt]:
```suggestion
be connected to the board for flashing. You can use JTAG or STLINK programmer.
```

File: boards/im880b/doc.txt [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+ *    - JTCK: PA_14 / P37
+ *    - JTMS: PA_13 / P34
+
+ *  STLINK:
+ *    - NRST: NRST / P7
+ *    - SWCLK: PA_14 / P37
+ *    - SWDAT: PA_13 / P34
+
+### STM32 Loader
+To flash RIOT on the board, after connection the UART-USB bridge, just run:
```
Comment (aabadie) [inline boards/im880b/doc.txt]:
```suggestion
To flash RIOT on the board, after connecting the UART-USB bridge, just run:
```

File: boards/im880b/doc.txt [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+```
+BOARD=im880b make flash
+```
+This uses the stm32loader script to erase the memory and flash it interfacing
```
Comment (aabadie) [inline boards/im880b/doc.txt]:
```suggestion
This uses the stm32loader script to erase the memory and flash it 
```

File: boards/im880b/include/board.h [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name    xtimer configuration
+ * @{
+ */
+#define XTIMER_BACKOFF              (11)
+#define XTIMER_OVERHEAD             (6)
```
Comment (aabadie) [inline boards/im880b/include/board.h]: Is this macro still needed with the new xtimer rework ?

File: boards/im880b/include/board.h [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+#define XTIMER_WIDTH                (16)
+/** @} */
+
+/**
+ * @name    sx1276 configuration
```
Comment (aabadie) [inline boards/im880b/include/board.h]:
```suggestion
 * @name    sx1272 configuration
```

File: boards/im880b/include/gpio_params.h [hunk 10]
```diff
@@ -0,0 +1,34 @@
+/*
```
Comment (aabadie) [inline boards/im880b/include/gpio_params.h]: This file is empty, it could be removed.

**Comment by fjmolinas** [comment]
Rebased and comments addressed

**Comment by fjmolinas** [comment]
There might be other stuff that is not standard anymore, will trigger a build to check.

**Comment by benpicco** [comment]
Some new tests were added that don't fit on the board.

**Comment by fjmolinas** [comment]
Everything is passing now, @kYc0o @aabadie can I squash?

**Comment by fjmolinas** [comment]
@aabadie can I squash when rebasing?

**Comment by aabadie** [comment]
> can I squash when rebasing?

Yes please

**Comment by aabadie** [comment]
nit: there a typo in 161dab0: `s/memmory/memory`

**Comment by fjmolinas** [comment]
FIxed typo and ammended insufficient memmory list

**Comment by aabadie** [comment]
> FIxed typo and ammended insufficient memmory list

Do you mean _memory_ (and _amended_) ? :laughing:

**Comment by benpicco** [comment]
@aabadie have your comments been addressed? 😉

**Comment by fjmolinas** [comment]
Thanks for the review!

**Pull request merged**